### DISCUSSION TITLE
Server should allow hostname configuration with standalone mode

### DISCUSSION THREAD

**Discussion by makkeu** [vercel/next.js#44626] [Ideas]
### Describe the feature you'd like to request

When building with `output: 'standalone'`, generated server.js has hardcoded 'localhost' as hostname. It should be made configurable via env variable, similar to `PORT`. `next start` has `--hostname` argument, but same isn't possible with generated `server.js`.

### Describe the solution you'd like

I'd like to start server using `HOSTNAME=x.x.x.x` env variable, and running generated standalone `server.js` would listen to that hostname.

### Describe alternatives you've considered

This prevents us to use standalone mode, and we must start the server using `next start` command.

**Comment by balazsorban44** [comment]
Opened a PR: https://github.com/vercel/next.js/pull/44627

For now, you can add a post-build script that replaces `hostname: 'localhost'` in the `server.js` file.

**Comment by WahidN** [comment]
This still doesn't work with an .env file. 

I have a .env file in the standalone map with HOSTNAME but it still uses '0.0.0.0'

**Comment by lexkrstn** [comment]
The server ignores the `HOSTNAME` environment variable even when running with the Next executor. I had to hardcode it in my PM2 config to pass the hostname via the argument:
```js
module.exports = {
  apps: [
    {
      script: 'node_modules/next/dist/bin/next',
      args: "start -H 127.0.0.1",  // <-- WORKS
      env: {
        HOSTNAME: "127.0.0.1", // <-- WON'T WORK
      },
    },
  ],
};
```
Given that `PORT` and other settings are loaded from the `.env` file and only `HOSTNAME` is failing, I'd argue this is more of a bug rather than an "Idea," which is how it’s currently categorized..

**Comment by rmlevangelio** [comment]
Facing same issue with @lexkrstn, when deployed in production, I can see that it uses the pod name for the network instead of the IP address

<img width="413" height="84" alt="image" src="https://github.com/user-attachments/assets/8c7f0576-a701-4f5b-8c50-5647c2b92c0b" />
